### PR TITLE
fix: export cjs and es

### PR DESCRIPTION
**Pull request by simllll**
as commented in #1266, this one would restore the old behaviour and introduces a new entrypoint for es modules.

for CJS (javascript):
`const Agenda = require('agenda');`

or for module (typescript):
`import Agenda from 'agenda/es'`

This is a "another" breaking change though, as a named import would not work with the default entrypoint anymore (didn't work before, but did now in the patch releases)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,19 @@ Install via NPM
...
+
+for regular javascript code, just use the default entrypoint
+```js
+const Agenda = require('agenda');
+```
+
+For Typescript, Webpack or other module imports, use `agenda/es` entrypoint:
+e.g.
+```ts
+import { Agenda } from 'agenda/es';
```
Comment (leonardlin) [inline README.md:73]: I would make a note that people migrating from @types/agenda should change their imports 
from: import Agenda from 'agenda'
to: import { Agenda} from 'agenda/es'

**Review by leonardlin — DISMISSED** [review]
I'm not an expert on module/exports etc. therefore I yield to other people.
But I added note for people migrating from @types/agenda

File: README.md (hunk 1 above)
Comment (simllll) [reply README.md:73]: added it to the readme

**Review by leonardlin — DISMISSED** [review]

**Comment by thebestnom** [comment]
Couldn't you just add 
```ts
export = Agenda
```
In index.ts?

**Comment by simllll** [comment]
> Couldn't you just add
> 
> ```ts
> export = Agenda
> ```
> 
> In index.ts?

Unfortunately not, this would overwrite all other exports and would therefore not work correctly. It would only fix the plain JS includes, but this does my PR too.

**Review by koresar — APPROVED** [review]

**Comment by simllll** [comment]
Thanks for the additional fixes and tests! Lgtm too, feel free to merge

**Pull request merged**